src/main/mw.c

```diff
@@ -686,6 +690,110 @@ void processRx(void)
 
 }
 
+void getArmingChannel(modeActivationCondition_t *modeActivationConditions, uint8_t *armingChannel) {
+    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
+        modeActivationCondition_t *modeActivationCondition = &modeActivationConditions[index];
+    	if (modeActivationCondition->modeId) {
```

Comment:
This will find last non-zero modeId (and BOXARM is zero ... )

maybe : `modeActivationCondition->modeId == BOXARM` ?